stacker/stack.py

```diff
@@ -114,6 +114,11 @@ def requires(self):
         for lookup in self.lookups:
             if lookup.type == OUTPUT_LOOKUP_TYPE_NAME:
                 d = deconstruct(lookup.input)
+                if d.stack_name == self.name:
+                    message = (
+                        "Lookup: \"%s\" is a circular reference to: %s"
```

Comment:
Can we provide the stack & variable that this lookup happens in as part of this error?  Might make it easier to chase down what is actually broken in some cases.  Especially with anchors and what not.